libs/ledger/src/transaction_status_cache.cpp

```diff
@@ -27,54 +27,55 @@ using fetch::generics::MilliTimer;
 namespace fetch {
 namespace ledger {
 
-char const *ToString(TransactionStatus status)
+char const *ToString(eTransactionStatus status)
 {
   char const *text = "Unknown";
 
   switch (status)
   {
-  case TransactionStatus::UNKNOWN:
+  case eTransactionStatus::UNKNOWN:
     break;
-  case TransactionStatus::PENDING:
+  case eTransactionStatus::PENDING:
     text = "Pending";
     break;
-  case TransactionStatus::MINED:
+  case eTransactionStatus::MINED:
     text = "Mined";
     break;
-  case TransactionStatus::EXECUTED:
+  case eTransactionStatus::EXECUTED:
     text = "Executed";
     break;
-  case TransactionStatus::SUBMITTED:
+  case eTransactionStatus::SUBMITTED:
     text = "Submitted";
     break;
   }
 
   return text;
 }
 
-TransactionStatus TransactionStatusCache::Query(Digest digest) const
+TransactionStatusCache::TxStatus TransactionStatusCache::Query(Digest digest) const
 {
-  TransactionStatus status{TransactionStatus::UNKNOWN};
+  TxStatus status;
 
   {
     FETCH_LOCK(mtx_);
 
     auto const it = cache_.find(digest);
     if (cache_.end() != it)
     {
-      status = it->second.status;
+      status = it->second;
     }
   }
 
   return status;
 }
 
-void TransactionStatusCache::Update(Digest digest, TransactionStatus status, Timepoint const &now)
+void TransactionStatusCache::Update(Digest digest, eTransactionStatus status,
+                                    ContractExecutionResult const execRes, Timepoint const &now)
```

Comment:
Style 🚓  - `exec_result`?

Also I think that we want to remove the `const` in order to make the move useful

Reply:
I also think I would probably separate the two types of update call (high level and execution level) into two unique methods and sort of the details internally.